apps/docs/app/routes/actions/user-preferences.tsx

```diff
@@ -0,0 +1,29 @@
+import { ActionFunction, json, LoaderFunction, redirect } from "remix";
+import { isUserPreferences } from "~/features/user-preferences/UserPreferencesContext";
+import { getUserPreferenceSession } from "~/utils/userPreferences.server";
+
+export const action: ActionFunction = async ({ request }) => {
```

Comment:
This is a so-called resource route, or an API route of sorts. It's called with as a post method, and creates the cookie as a response.